src/packet.c

```diff
@@ -1127,16 +1127,6 @@ _libssh2_packet_require(LIBSSH2_SESSION * session, unsigned char packet_type,
                                       match_ofs, match_buf, match_len);
             state->start = 0;
             return ret;
-        } else if (ret == 0) {
```

Comment:
I agree that this code was pointless because you could only get if if the socket was disconnected, in which case who cares about the timeout.  But the timeout should not be applied to the LIBSSH2_ERROR_EAGAIN case, as done in _packet_requirev.

As you say though, it should be possible to implement _packet_require in terms of _packet_requirev.